code/game/objects/items/robot/robot_items/robot_furnace.dm

```diff
@@ -0,0 +1,100 @@
+//Totally not copypasted from its machinery version stop asking so many questions
+#define PORTOSMELTER_MINIMUM_POWER_REQUIRED 100
+#define PORTOSMELTER_POWER_DRAIN rand(50,100)
+
+/obj/item/weapon/storage/bag/ore/furnace
+	name = "portable ore processor"
+	icon_state = "portsmelter"
+	actions_types = list(/datum/action/item_action/toggle_furnace)
+	var/event_key = null
+	var/datum/materials/ore
+	var/list/recipes[0]
+	var/on = FALSE
+	var/sheets_per_tick = 100
+
+/datum/action/item_action/toggle_furnace
+	name = "Toggle Ore Furnace"
+
+/datum/action/item_action/toggle_furnace/Trigger()
+	var/obj/item/weapon/storage/bag/ore/furnace/F = target
+	var/mob/user = usr
+
+	if(!usr)
```

Comment:
Action buttons have an `owner` variable which is almost invariably the user
I'm not sure if this use of `usr` is necessary or wise, but apparently this is copypaste